common-npm-packages/azure-arm-rest-v2/azure-arm-common.ts

```diff
@@ -296,7 +296,17 @@ export class ApplicationTokenCredentials {
         // proxy usage
         const agentProxyURL = tl.getVariable("agent.proxyurl") ? new URL(tl.getVariable("agent.proxyurl")) : null;
         const agentProxyBypassHosts = tl.getVariable("agent.proxybypasslist") ? JSON.parse(tl.getVariable("agent.proxybypasslist")) : null;
-        const shouldProxyBypass = agentProxyBypassHosts?.includes(new URL(authorityURL).host);
+        let shouldProxyBypass = false;
+
+        const authorityHost = new URL(authorityURL).host;
+        for(var i=0; agentProxyBypassHosts && i < agentProxyBypassHosts.length; i++) {
+            // same test logic with typed-rest-client
+            if(agentProxyBypassHosts[i] && new RegExp(agentProxyBypassHosts[i], 'i').test(authorityHost)) {
```

Comment:
1) Formatting - please put a space between if and opening bracket
2) when using regex please add a comment to reflect - what this is supposed to catch, since in most case regex is hard to understand without a context